sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -1,20 +1,742 @@
 package io.sentry.android.core;
 
+import static android.content.Context.ACTIVITY_SERVICE;
+
+import android.app.ActivityManager;
 import android.content.Context;
-import io.sentry.core.EventProcessor;
-import io.sentry.core.SentryEvent;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageInfo;
+import android.content.res.AssetManager;
+import android.content.res.Configuration;
+import android.net.ConnectivityManager;
+import android.os.*;
+import android.provider.Settings;
+import android.util.DisplayMetrics;
+import io.sentry.core.*;
+import io.sentry.core.protocol.*;
+import io.sentry.core.util.Objects;
+import java.io.*;
+import java.util.*;
 
 public class DefaultAndroidEventProcessor implements EventProcessor {
   Context context;
+  SentryOptions options;
 
-  public DefaultAndroidEventProcessor(Context context) {
-    if (context == null) throw new IllegalArgumentException("The application context is required.");
-    this.context = context.getApplicationContext();
+  public DefaultAndroidEventProcessor(Context context, SentryOptions options) {
+    this.context =
+        Objects.requireNonNull(
+            context != null ? context.getApplicationContext() : null,
+            "The application context is required.");
+    this.options = Objects.requireNonNull(options, "The SentryOptions is required.");
   }
 
   @Override
   public SentryEvent process(SentryEvent event) {
-    // TODO: Do cool android things.
+    if (event.getSdkVersion() == null) {
+      SdkVersion sdkVersion = new SdkVersion();
+      sdkVersion.setName("android");
+      // version, don't we have getRelease() for that?
+      // packages, which packages?
+
+      event.setSdkVersion(sdkVersion);
+    }
+    if (event.getUser() == null) {
+      event.setUser(getUser());
+    }
+
+    PackageInfo packageInfo = getPackageInfo();
+    if (packageInfo != null) {
+      if (event.getRelease() == null) {
+        event.setRelease(packageInfo.packageName + "-" + packageInfo.versionName);
```

Comment:
I kept the same as before, so it looks like `io.sentry.sample-1.0`, but we also have the same info with `contexts.app.identifier` and `contexts.app.identifier.version` and more

Reply:
I see so it would be a non breaking change to keep it like this.

Btw `release` will be automatically index in Sentry. The data in `App` is not so it's OK to have it "duplicated".

Often in other SDKs (other than Java) what ppl use for Release is the git commit SHA. Or the app numeric version like _1.0.0_. I haven't seen it yet as a combination of appname-version.
@HazAT Do you have an opinion here?

Reply:
For now I would leave it like this. Once we revamp the release system we will change this for sure.